embed parser application codebase in winestyle-parser-service

baby-benz opened the pull request:
Parser services have been implemented. Communication with the database occurs through the business logic layer. The design of interfaces based on REST Api principles has begun. A generic controller exception handler has been added.

baby-benz committed:
embed parser application codebase in winestyle-parser-service

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/ParseController.java:
```diff
@@ -0,0 +1,22 @@
+    private final IParserService parserService;
+
+    @RequestMapping(value = "{alcohol}", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
```
Вы же и так внизу возвращаете "ok". Зачем это?

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/ParseController.java:
```diff
@@ -0,0 +1,22 @@
+public class ParseController {
+    private final IParserService parserService;
+
+    @RequestMapping(value = "{alcohol}", method = RequestMethod.POST)
```
Вы, кстати,  можете заюзать @PostMapping аннотацию, она более новая

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/ParserService.java:
```diff
@@ -0,0 +1,304 @@
+    }
+
+    //At 00:00; every day
+    @Scheduled(cron = "0 0 0 * * *") // second, minute, hour, day of month, month, day(s) of week(0-6)
```
Вынесите лучше в проперти

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/WineService.java:
```diff
@@ -0,0 +1,101 @@
+            wine.setColor(wineDto.getColor());
+        if (wineDto.getRegion() == null) {
+            wine.setRegion("noRegion");
+        } else
```
Лучше все-таки юзать скобки. Вы в ифе на одну строчку это делаете, а так странно выглядит

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/DocumentService.java:
```diff
@@ -0,0 +1,37 @@
+                log.error("Couldn't get a connection to website! ", ex);
+            } // Берем страничку html
+        }
+        Thread.sleep(600);
```
Это зачем?

baby-benz replied:
Это временное решение, чтобы избежать IP-блокинга за слишком большое количество запросов. В будущем добавлю пул потоков с менеджером и в отдельном потоке будет исполняться метод по забору страниц, чтобы это время использовать с пользой и главный поток не простаивал.

shinehlina reviewed (changes requested):
Только очень минорные штуки)

NShmendyuk committed:
added cron expression in application properties, check service business on scheduler

baby-benz committed:
large scale project restructuring
change getElementBySmth to selectFirst for performance increase purposes
implement builder pattern by adding additional @Builder and constructor annotations
tell JPA Hibernate to autogenerate table by adding spring.jpa.hibernate.ddl-auto=update to .properties file

baby-benz committed:
Merge remote-tracking branch 'origin/feature/scheduler-upgrade' into dev

# Conflicts:
#	src/main/java/com/wine/to/up/winestyle/parser/service/service/ParserService.java

baby-benz committed:
minor scheduler and entity properties fixes

baby-benz committed:
fix initial condition in parseByPage loop

Commit:
Javadoc added

baby-benz committed:
Merge pull request #1 from Gashad0kuro/dev

Javadoc added

kanootoko committed:
REST API update
Add API to get all wines, wine by id, CSV file with all wines (needs testing), wine with given columns
Change path of API to have prefix "/winestyle"
Change CSVUtility format of output

baby-benz committed:
Merge pull request #2 from baby-benz/kanootoko

REST API update

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/MainController.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    @GetMapping("/wine/with_fields/{id}")
+    public ResponseEntity<Map<String, Object>> getParsedWineWithFields(@PathVariable long id,
```
Напишите здесь javadoc, что это за fields такие

shinehlina replied:
Потому что сейчас я не очень понимаю, зачем это)

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,55 @@
+     * @param request Текущий запрос
+     */
+    @ExceptionHandler(ServiceIsBusyException.class)
+
```
Пробел

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/domain/entity/Wine.java:
```diff
@@ -0,0 +1,120 @@
+     */
+    @Override
+    public String toString(){
+        return url + " "
```
`@ToString`

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/implementation/WineService.java:
```diff
@@ -0,0 +1,78 @@
+    public Wine getWineByID(long id) {
+        Optional<Wine> wine = wineRepository.findById(id);
+        if (wine.isEmpty()) {
+            return null;
```
В целом не очень хорошая практика null возвращать. Лучше либо Optional, либо бросать исключение. Можете поправить.

shinehlina commented on src/main/resources/application.properties:
```diff
@@ -12,3 +16,5 @@ default.kafka.bootstrap-server=localhost:29092
 
 # flyway properties - https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html#data-migration-properties
+
+cron.expression=0 0 0 * * *
```
Это как расшифровывается? Напишите сверху

shinehlina reviewed (changes requested):
В целом мне понравилось)

NShmendyuk committed:
added exception when no entity found by id (no response body yet), added description for scheduler cron

baby-benz committed:
make HttpStatus code and body message more informative
replace custom Exceptions constructors with annotations

baby-benz committed:
add javadoc to getParsedWineWithFields
replace Wine toString() method with ToString annotation

baby-benz committed:
replace ParserService's and WineService's constructor with Constructor annotation
fix getParsedWineWithFields
minor refactoring

baby-benz committed:
make CSVUtility methods invocation through the interface, not directly calling class instance methods.

baby-benz committed:
turn CSVUtility into a full static utility-class using @UtilityClass annotation
fix toCsvFile method header issue

kanootoko committed:
Merge pull request #3 from baby-benz/fixes

Fixes
Refactoring (interfaces without 'I' prefix)
Change database name (postgres -> winestyle)

kanootoko committed:
Version 0.2.0
Add parser integration with Kafka
Remove all metions of demo service
Small refactoring

Commit:
updated javadoc

baby-benz committed:
add sparkling wines parser support
global project restructuring
small naming issues fixes

baby-benz committed:
deeper refactoring

baby-benz committed:
ProductPageParser annotation fix

baby-benz committed:
Builder classes bugfix

baby-benz committed:
add handling of one more condition in parseTypeColorSugar and parseColorAndSugar
add sparkling drinks crop year field support

baby-benz committed:
fix parseCropYear title splitting

baby-benz committed:
fix parseCropYear title splitting

shinehlina commented on pom.xml:
```diff
@@ -73,6 +73,29 @@
+        <dependency>
+            <groupId>org.jsoup</groupId>
+            <artifactId>jsoup</artifactId>
+            <version>1.13.1</version>
```
Давайте все версии в проперти

shinehlina commented on pom.xml:
```diff
@@ -55,7 +55,7 @@
         </dependency>
 
-        <!-- Recommended-->
+        <!-- Recommended -->
```
Апните еще версию

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/KafkaController.java:
```diff
@@ -27,7 +27,7 @@
  */
 @RestController
-@RequestMapping("/kafka")
+@RequestMapping("/winestyle/kafka")
```
Можете удалить этот контроллер вообще

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/MainController.java:
```diff
@@ -0,0 +1,93 @@
+     * @param id id вина в базе данных.
+     * @param fieldsList список запрашиваемых полей.
+     * @return HTTP-статус 200(ОК) и вино с запрошенными полями в теле ответа.
+     * @throws NoEntityException если искомое вино не найдено.
```
Отформатируйте alt ctrl L

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/MessageController.java:
```diff
@@ -15,9 +14,8 @@
+@RequestMapping("/winestyle/message")
 @Validated
-@Slf4j
 public class MessageController {
```
Этот тоже можете удалить, раз уже написали свой

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,68 @@
+     * @param ex      Целевое исключение
+     * @param request Текущий запрос
+     */
+    @ExceptionHandler({ServiceIsBusyException.class, NoEntityException.class})
```
Почему вы вам не сделать хендлер отдельно для ServiceIsBusyException, отдельно для NoEntityException и просто для Exception. Такая проверка не нужна, если есть рекомендуемый способ ` if (ex instanceof ServiceIsBusyException) {`

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,68 @@
+        return handleExceptionInternal(ex, new ApiError(errors), headers, status, request);
+    }
+
+    protected ResponseEntity<ApiError> handleNoEntityException(NoEntityException ex,
```
protected?

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/controller/exception/ServiceIsBusyException.java:
```diff
@@ -0,0 +1,21 @@
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+/**Класс  ошибка, который кидается
```
ctrl alt L на весь проект)

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/domain/entity/Message.java:
```diff
@@ -22,4 +22,4 @@
         this.content = content;
     }
-}
+}
```
Зачем вам этот класс уже, если у вас уже свои энтити

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/logging/WinestyleParserServiceNotableEvents.java:
```diff
@@ -2,6 +2,11 @@
 import com.wine.to.up.commonlib.logging.NotableEvent;
 
+/**
+ * Перечисление шаблонов сообщений в/от Kafka
```
Ну это не совсем так. Это метрики в Промитиус будут потом

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/implementation/WinestyleParserService.java:
```diff
@@ -0,0 +1,467 @@
+
+        int pages = documentService.pagesNumber(mainDoc);
+
+        for (int i = 2; i <= pages; i++) {
```
Почему 2?

shinehlina commented on src/main/resources/application.properties:
```diff
@@ -1,14 +1,20 @@
 management.endpoint.prometheus.enabled=true
 
-spring.datasource.url=jdbc:postgresql://${S_POSTGRES_HOST:localhost}:5432/postgres
+spring.jpa.hibernate.ddl-auto=update
```
Давайте лучше через миграции создавать, как Андрей показывал. И тогда это validate

baby-benz committed:
final project refactoring
parsing optimizations

baby-benz committed:
final project optimizations
primary codebase for Image scraping

baby-benz committed:
fix Alcohol entity JavaDoc

baby-benz committed:
add database table to collect errors on saving data to main database
update migration sql-file
upgrade parser to process fields not belonginging to HTML-elements

baby-benz committed:
minor changes in REST-endpoints names

baby-benz committed:
fix flyway migration SQL-file

baby-benz committed:
parser no type property bugfix
fix flyway migration SQL-file

baby-benz committed:
fix type property parsing

baby-benz committed:
remove redundant ResponseEntity return type from MainController

kanootoko committed:
Merge pull request #3 from microservices-course-itmo/feature/refactoring

Refactoring. Sparkling drinks support

baby-benz committed:
upgrade wine parser for different alcohol types identification
fix migration SQL-file
upgrade ErrorOnSaving class to save exception messages
remove @Acessors(fluent = true) for correct domain mapping

baby-benz committed:
Merge branch 'feature/refactoring' into sprint-5

kanootoko committed:
Version 0.2.0 - merged with kafka_integration

Commit:
javadoc update sp5

kanootoko committed:
Version 0.5.0
Finally merge kafka integration branch
Change most of the "wine" enterences to "alcohol"
Add logger
Try to change database to winestyle via properties
Fix javadoc a bit
Update version

kanootoko committed:
Merge pull request #4 from microservices-course-itmo/logger

Version 0.5.0
    Finally merge kafka integration branch
    Change most of the "wine" enterences to "alcohol"
    Add logger
    Try to change database to winestyle via properties
    Fix javadoc a bit
    Update version

baby-benz committed:
add url handler to NoEntityException
minor fixes

baby-benz committed:
add more wine types support
fix parsing duration timer
fix NoEntityException throwing while parsing

baby-benz committed:
fix non-alcoholic products kafka sending
update error_on_saving database

baby-benz committed:
fix migration SQL-file

baby-benz committed:
update properties file

baby-benz committed:
drag lib versions to the properties tag

baby-benz committed:
fix endpoint naming

baby-benz committed:
process null conditions at entity creation

baby-benz committed:
update field size reflecting changed requirements

baby-benz committed:
make StatusService method signature clearer

baby-benz committed:
fix small typos

baby-benz committed:
fix regex matching groups

baby-benz committed:
make logging before making attempt to save in error repository

baby-benz committed:
adapt database to changed requirements

baby-benz committed:
adapt database to changed requirements #2

baby-benz committed:
fix parsing timer issue

baby-benz committed:
add wines without type processing

baby-benz committed:
final db field size adjustments
fix kafka builder

baby-benz committed:
bugfix wine type parsing

baby-benz committed:
exclude children's champagne from results

baby-benz committed:
children's champagne is no longer supported(
make type parsing more accurate

baby-benz committed:
fix sparkling url

baby-benz committed:
delete ambiguous casting

baby-benz committed:
prepare project for deployment

baby-benz committed:
add new interface for repository services
fix getAllXyz() family methods

baby-benz committed:
fix AlcoholRepository queries

Commit:
Javadoc updated

Commit:
Javadoc updated

shinehlina commented on pom.xml:
```diff
@@ -16,11 +15,14 @@
 
     <properties>
-        <current-version>0.1.0</current-version>
+        <current-version>0.5.0</current-version>
```
Мы пока считаем, что версии только в деве, так что это тогла 0.2.0

kanootoko replied:
> Мы пока считаем, что версии только в деве, так что это тогда 0.2.0
>
Ну, мне кажется, что у нас должен быть релиз какой-то после каждого спринта, и чтобы было удобнее понимать, какая версия была когда готова, нужно использовать наименование "номер версии = номер спринта". Это был пятый спринт. Ну, подумаешь, в деве скипнутся 0.2.0-0.4.0, зато дальше будет проще

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/ServiceApplication.java:
```diff
@@ -2,16 +2,20 @@
 
 @SpringBootApplication
-@ComponentScan("com.wine.to.up")
+@ComponentScan({"com.wine.to.up", "com.wine.to.up.winestyle.parser.service.service"})
```
com.wine.to.up включает в себя com.wine.to.up.winestyle.parser.service.service

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/ServiceApplication.java:
```diff
@@ -2,16 +2,20 @@
-@ComponentScan("com.wine.to.up")
+@ComponentScan({"com.wine.to.up", "com.wine.to.up.winestyle.parser.service.service"})
+@EntityScan("com.wine.to.up.winestyle.parser.service.domain")
+@EnableJpaRepositories("com.wine.to.up.winestyle.parser.service.repository")
```
Это тоже, кажеься, лишнее

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/ServiceApplication.java:
```diff
@@ -2,16 +2,20 @@
 @SpringBootApplication
-@ComponentScan("com.wine.to.up")
+@ComponentScan({"com.wine.to.up", "com.wine.to.up.winestyle.parser.service.service"})
+@EntityScan("com.wine.to.up.winestyle.parser.service.domain")
```
Бещ этого не работает?

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/implementation/document/DocumentService.java:
```diff
@@ -0,0 +1,54 @@
+            log.info("Parsing: {}?page={}", alcoholUrl, nextPageNumber);
+            return scrapingService.getJsoupDocument(alcoholUrl + "?page=" + nextPageNumber);
+        } else {
+            return null;
```
null возвращаете

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/implementation/document/DocumentService.java:
```diff
@@ -0,0 +1,54 @@
+            String pagesNumber = mainDoc.selectFirst("#CatalogPagingBottom li:last-of-type").text();
+            return Integer.parseInt(pagesNumber);
+        } catch (NullPointerException ex) {
+            return 1;
```
Залогируйте, почему эта ошибка может быть. И почему вообще 1 возвращаете, а не 0?

shinehlina replied:
Все равно неочевидно, но ок

shinehlina commented on src/main/java/com/wine/to/up/winestyle/parser/service/service/implementation/document/ScrapingService.java:
```diff
@@ -0,0 +1,40 @@
+                log.error("Couldn't get a connection to website!", ex);
+            } // Берем страничку html
+            catch (IOException e) {
+                e.printStackTrace();
```
Логгер

shinehlina reviewed (changes requested):
только по возможности не пуште в pr ничего нового. Пусть pr за спринт все-таки будет

baby-benz committed:
remove superfluous DocumentService.java business-logic layer

baby-benz committed:
replace stack trace printing by logging

baby-benz committed:
remove redundant annotations
simplify ComponentScan annotation

baby-benz committed:
Merge remote-tracking branch 'origin/sprint-5' into sprint-5

baby-benz committed:
ScrapingService's javadoc minor fix

baby-benz committed:
fix while loop making it run through all pages

baby-benz committed:
fix while loop making it run through all pages
